Content.Server/Storage/EntitySystems/DumpableSystem.cs

```diff
@@ -73,15 +94,38 @@ private void AddUtilityVerbs(EntityUid uid, DumpableComponent dumpable, GetVerbs
             if (!TryComp<ServerStorageComponent>(uid, out var storage) || storage.StoredEntities == null || storage.StoredEntities.Count == 0)
                 return;
 
-            if (HasComp<DisposalUnitComponent>(args.Target))
+            var addverbstorage = false;
+            if (TryComp<MaterialStorageComponent>(args.Target, out var materialStorage))
+            {
+                foreach (var mat in materialStorage!.EntityWhitelist!.Tags!)
+                {
+                    foreach (var serv in storage!.Whitelist!.Tags!)
+                    {
+                        if (serv == mat)
+                        {
+                            addverbstorage = true;
+                            break;
+                        }
+                    }
+                    if (addverbstorage)
+                        break;
+                }
+            }
```

Comment:
Same here. Additionally you should put this in another method rather than copying it.